traits/ctraits.c

```diff
@@ -4253,6 +4259,37 @@ _trait_comparison_mode(trait_object *trait, PyObject *args)
     return Py_None;
 }
 
+/*-----------------------------------------------------------------------------
+|  getter for trait comparison mode
++----------------------------------------------------------------------------*/
+
+static PyObject *
+_get_trait_comparison_mode_int(trait_object *trait, PyObject *Py_UNUSED(ignored))
+{
+    int i_comparison_mode;
+
+    unsigned long compare_flag = trait->flags & TRAIT_COMPARE_MASK;
+
+    if (compare_flag == TRAIT_NO_VALUE_TEST) {
+        i_comparison_mode = 0;
+    }
+    else if (compare_flag == TRAIT_OBJECT_ID_TEST) {
+        i_comparison_mode = 1;
+    }
+    else {
+        i_comparison_mode = 2;
+    }
+
+    PyLongObject *comparison_mode = PyLong_FromLong(i_comparison_mode);
```

Comment:
nitpick: please could you move the `PyLongObject *comparison_mode` declaration to the top of the function? This used to matter until quite recently on Windows, and could conceivably cause build issues on some machines.

Reply:
Actually, you can lose the declaration entirely and simply do `return PyLong_FromLong(i_comparison_mode)`. That does the right thing both in the case of failure (returning `NULL`) and success.